modin/data_management/functions/groupby_function.py

```diff
@@ -272,6 +273,7 @@ def caller(
             or isinstance(by, pandas.Grouper)
         ):
             by = try_cast_to_pandas(by, squeeze=True)
+            by = GroupBy.validate_by(by)
```

Comment:
[`GroupBy.validate_by`](https://github.com/modin-project/modin/blob/4abb53b4a04a66e461a2169f585e474113781be4/modin/data_management/functions/default_methods/groupby_default.py#L34) splits 2D 'by' into a list of 1D Series

Reply:
It would be really great to have these comments in the code. Nobody reads old closed PRs especially if comments are hidden by "Resolve conversation". All these insights will be lost if they aren't present in the code.

Reply:
I just retyped a shorter version of `GroupBy.validate_by` docstring in this comment for reviewers' convenience. In the code, it's your IDE's work to explain what the function is going by docstrings.

Reply:
I side with @gshimansky here - if you feel that you need to add something so reviewers understand the code better this means that either the code needs to be made cleaner or those comments should be added to the diff. Do not rely on IDE to highlight - there are people who do not use IDEs or do not set up those to show docstrings or whatever.

Reply:
added